src/tree/updater_quantile_hist.h

```diff
@@ -273,9 +277,17 @@ class QuantileHistMaker: public TreeUpdater {
     std::shared_ptr<common::ColumnSampler> column_sampler_{
         std::make_shared<common::ColumnSampler>()};
 
-    std::vector<size_t> unused_rows_;
     // the internal row sets
     RowSetCollection row_set_collection_;
+    std::vector<uint16_t> node_ids_;
```

Comment:
Could you please try to limit the use of memory variables?

Reply:
yes. Deleted some of them